lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/swig/SwigUtils.scala

```diff
@@ -26,6 +26,13 @@ object SwigUtils extends Serializable {
       lightgbmlib.floatArray_setitem(colArray, ri._2.toLong, ri._1))
     colArray
   }
+
+  def byteArrayToNative(array: Array[Byte]): SWIGTYPE_p_unsigned_char = {
+    val colArray = lightgbmlib.new_byteArray(array.length)
+    array.zipWithIndex.foreach(ri => array.zipWithIndex.foreach(ri =>
```

Comment:
Could for-comprehension be used here? It could make this code easier to comprehend IMHO.

```
for {
  ri1 <- array.zipWithIndex
  ri2 <- array.zipWithIndex
} yield (...)
```

Looks weird repeating `array.zipWithIndex` that could reveal the real intent of the code.

Reply:
I think this was just a copy/paste bug. :)  removed and testing